### PR TITLE
[DDW-455] Lock file mechanism UX changes

### PR DESCRIPTION
**Pull request by daniloprates**
This PR improves the lock file mechanism UX when the user tries to launch a secondary Daedalus instance. Now it focuses the first window after exiting the second one.

---

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn run flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn run storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

**Comment by daniloprates** [comment]
@nikolaglumac although I didn't add any mechanism to control the network source, it seems to behave exactly like the existing part, which prevents the second instance.

**Comment by nikolaglumac** [comment]
@DominikGuzei do you see any issues with this PR in relation to the changes you introduce in this PR? https://github.com/input-output-hk/daedalus/pull/1229

**Comment by DominikGuzei** [comment]
No, this looks great 👍 my changes in #1229 work exactly the same way as before with the lock files.

**Comment by nikolaglumac** [comment]
@DominikGuzei thanks!
Before we merge this one @daniloprates needs to run some more testing.
Once he is done with this he will report the test results here in the PR.

**Comment by daniloprates** [comment]
I have tested on Windows and it works as expected: 

* Opening 2 instances of the same network will bring the first to focus after the error message. 

* Opening 2 instances of different networks will not trigger the error nor the focus.

File: source/main/index.js [hunk 1]
```diff
@@ -81,9 +81,6 @@ app.on('ready', async () => {
   try {
     await acquireDaedalusInstanceLock();
   } catch (e) {
-    const dialogTitle = 'Daedalus is unable to start!';
-    const dialogMessage = 'Another Daedalus instance is already running.';
-    dialog.showErrorBox(dialogTitle, dialogMessage);
     app.exit(1);
```
Comment (nikolaglumac) [inline source/main/index.js]: @daniloprates I am not entirely sure that we want this `app.exit()` call here. Or that we event need try catch at all.
Please test on Windows once the build is ready to confirm the feature works as expected in terms of:
- having the first instance window being shown and focused
- having the second Daedalus instance process killed (confirmed by looking at active processes in the TaskManager).
Thanks!

**Comment by nikolaglumac** [comment]
Closing this PR unmerged as the functionality is merged into this PR: https://github.com/input-output-hk/daedalus/pull/1226

**Pull request closed without merging**